src/targets/maven.ts

```diff
@@ -0,0 +1,334 @@
+import { TargetConfig } from '../schemas/project_config';
+import {
+  BaseArtifactProvider,
+  RemoteArtifact,
+} from '../artifact_providers/base';
+import { BaseTarget } from './base';
+import { homedir } from 'os';
+import { basename, join, parse } from 'path';
+import { promises as fsPromises } from 'fs';
+import { checkExecutableIsPresent, extractZipArchive } from '../utils/system';
+import { retrySpawnProcess } from '../utils/async';
+import { withTempDir } from '../utils/files';
+import { checkEnvForPrerequisite } from '../utils/env';
+import { ConfigurationError } from '../utils/errors';
+import { stringToRegexp } from '../utils/filters';
+
+const GRADLE_PROPERTIES_FILENAME = 'gradle.properties';
+
+/**
+ * Default gradle user home directory. See
+ * https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_environment_variables
+ */
+const DEFAULT_GRADLE_USER_HOME = join(homedir(), '.gradle');
+
+export const targetSecrets = ['OSSRH_USERNAME', 'OSSRH_PASSWORD'] as const;
+type SecretsType = typeof targetSecrets[number];
+
+export const targetOptions = [
+  'gradleCliPath',
+  'mavenCliPath',
+  'mavenSettingsPath',
+  'mavenRepoId',
+  'mavenRepoUrl',
+] as const;
+type OptionsType = typeof targetOptions[number];
+
+type AndroidFields = {
+  android: {
+    distDirRegex: RegExp;
+    fileReplaceeRegex: RegExp;
+    fileReplacerStr: string;
+  };
+};
+
+type TargetSettingType = SecretsType | OptionsType;
+
+/**
+ * Config options for the "maven" target.
+ */
+export type MavenTargetConfig = Record<TargetSettingType, string> &
+  AndroidFields;
+
+/**
+ * Target responsible for uploading files to Maven Central.
+ */
+export class MavenTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'maven';
+  /** Target options */
+  public readonly mavenConfig: MavenTargetConfig;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.mavenConfig = this.getMavenConfig();
+    this.checkRequiredSoftware();
+  }
+
+  /**
+   * Returns the maven config with the required data (e.g. environment
+   * variables) for this target. If there's a configuration requirement missing,
+   * raises an error.
+   *
+   * @returns the maven config for this target.
+   */
+  private getMavenConfig(): MavenTargetConfig {
+    return {
+      ...this.getTargetSecrets(),
+      ...this.getOuterTargetSettings(),
+      ...this.getAndroidSettings(),
+    };
+  }
+
+  private getTargetSecrets(): Record<TargetSettingType, string> {
+    const secrets = targetSecrets.map(name => {
+      checkEnvForPrerequisite({ name });
+      return {
+        name,
+        value: process.env[name],
+      };
+    });
+    return this.reduceConfig(secrets);
+  }
+
+  private reduceConfig(
+    config: { name: string; value: string | undefined }[]
```

Comment:
Nit: prefer Array type instead of [] if the generic is not a primitive.
```suggestion
    config: Array<{ name: string; value: string | undefined }>
```

Reply:
I think a real improvement would come from extracting this type into an interface. Otherwise I don't see a huge difference in terms of readability.